lib/ast.js

```diff
@@ -191,17 +191,20 @@ util.methods(Symbols, {
     this.register('functions', name, fn);
   },
 
-  registerPath: function(parts, methods) {
+  registerPath: function(parts, isType, methods) {
+    isType = isType || 'Any';
     var p = {
       parts: parts,
+      isType: isType,
       methods: methods
     };
     this.register('paths', '/' + parts.join('/'), p);
   },
 
   registerSchema: function(name, derivedFrom, properties, methods) {
+    derivedFrom = derivedFrom || (Object.keys(properties).length > 0 ? 'Object' : 'Any');
```

Comment:
This is just to support:

type MyType {} === type MyType is Any {}
type MyType {prop: Foo} === type MyType is Object {prop: Foo}

Reason being, that you might describe an "empty" type, with just a validation expression.  We don't want to make it type Object as that adds additional validation that the storage is a structured Object (but it could be a rule for a scalar type).